Subject: [PATCH] temp update
---
Index: src/main/java/com/x_tornado10/lobby/Lobby.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/x_tornado10/lobby/Lobby.java b/src/main/java/com/x_tornado10/lobby/Lobby.java
--- a/src/main/java/com/x_tornado10/lobby/Lobby.java	(revision 4c37610bb705f2517786bec0a138383cf700b42e)
+++ b/src/main/java/com/x_tornado10/lobby/Lobby.java	(revision 2e946039d35b259313f26d7e82f53fa153aec287)
@@ -6,6 +6,7 @@
 import com.x_tornado10.lobby.commands.GrantRankCommand;
 import com.x_tornado10.lobby.commands.GrantRankCommandTabCompletor;
 import com.x_tornado10.lobby.commands.LobbyCommand;
+import com.x_tornado10.lobby.commands.SuffixChangeCommand;
 import com.x_tornado10.lobby.db.Database;
 import com.x_tornado10.lobby.listeners.JoinListener;
 import com.x_tornado10.lobby.listeners.LobbyListener;
@@ -114,6 +115,10 @@
             grantRank.setExecutor(new GrantRankCommand());
             grantRank.setTabCompleter(new GrantRankCommandTabCompletor());
         }
+        PluginCommand setSuffix = Bukkit.getPluginCommand("setsuffix");
+        if (setSuffix != null) {
+            setSuffix.setExecutor(new SuffixChangeCommand());
+        }
         lpAPI = LuckPermsProvider.get();
         Bukkit.getPluginManager().registerEvents(joinListener, this);
         Bukkit.getPluginManager().registerEvents(new LobbyListener(configMgr.getDoor()), this);
Index: src/main/java/com/x_tornado10/lobby/commands/SuffixChangeCommand.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/java/com/x_tornado10/lobby/commands/SuffixChangeCommand.java b/src/main/java/com/x_tornado10/lobby/commands/SuffixChangeCommand.java
--- a/src/main/java/com/x_tornado10/lobby/commands/SuffixChangeCommand.java	(revision 4c37610bb705f2517786bec0a138383cf700b42e)
+++ b/src/main/java/com/x_tornado10/lobby/commands/SuffixChangeCommand.java	(revision 2e946039d35b259313f26d7e82f53fa153aec287)
@@ -1,13 +1,70 @@
 package com.x_tornado10.lobby.commands;
 
+import com.x_tornado10.lobby.Lobby;
+import com.x_tornado10.lobby.utils.statics.Convertor;
+import net.luckperms.api.LuckPerms;
+import net.luckperms.api.model.user.User;
+import net.luckperms.api.model.user.UserManager;
+import net.luckperms.api.node.NodeType;
+import net.luckperms.api.node.types.PrefixNode;
+import net.luckperms.api.node.types.SuffixNode;
+import org.bukkit.ChatColor;
 import org.bukkit.command.Command;
 import org.bukkit.command.CommandExecutor;
 import org.bukkit.command.CommandSender;
+import org.bukkit.entity.Player;
 import org.jetbrains.annotations.NotNull;
 
 public class SuffixChangeCommand implements CommandExecutor {
+    private final Lobby plugin;
+    private LuckPerms lpAPI;
+    public SuffixChangeCommand() {
+        plugin = Lobby.getInstance();
+        lpAPI = plugin.getLpAPI();
+    }
     @Override
-    public boolean onCommand(@NotNull CommandSender commandSender, @NotNull Command command, @NotNull String s, @NotNull String[] strings) {
-        return false;
+    public boolean onCommand(@NotNull CommandSender commandSender, @NotNull Command command, @NotNull String s, @NotNull String[] args) {
+        if (commandSender instanceof Player p) {
+            if (plugin.checkGroup(p, "cs+") || plugin.checkGroup(p, "csp")) {
+                switch (args.length) {
+                    case 0 -> p.sendMessage(ChatColor.RED + "Please provide a new suffix!");
+                    case 1 -> {
+                        if (lpAPI == null) lpAPI = plugin.getLpAPI();
+                        if (lpAPI == null) return false;
+                        UserManager usrMgr = lpAPI.getUserManager();
+                        User usr = usrMgr.getUser(p.getName());
+                        if (usr == null) return false;
+                        for (SuffixNode snode : usr.getNodes(NodeType.SUFFIX)) {
+                            usr.data().remove(snode);
+                        }
+                        usr.data().add(SuffixNode.builder(ChatColor.RESET + args[0], 5).build());
+                        usrMgr.saveUser(usr);
+                        lpAPI.runUpdateTask();
+                    }
+                    default -> p.sendMessage(ChatColor.RED + "You provided too many arguments!");
+                }
+            } else {
+                p.sendMessage(ChatColor.RED + "You don't have permission to use this command!");
+            }
+        } else {
+            switch (args.length) {
+                case 0,1 -> plugin.getLogger().info("Please provide all necessary arguments!");
+                case 2 -> {
+                    if (lpAPI == null) lpAPI = plugin.getLpAPI();
+                    if (lpAPI == null) return false;
+                    UserManager usrMgr = lpAPI.getUserManager();
+                    User usr = usrMgr.getUser(args[0]);
+                    if (usr == null) return false;
+                    for (SuffixNode snode : usr.getNodes(NodeType.SUFFIX)) {
+                        usr.data().remove(snode);
+                    }
+                    usr.data().add(SuffixNode.builder(ChatColor.RESET + args[0], 5).build());
+                    usrMgr.saveUser(usr);
+                    lpAPI.runUpdateTask();
+                }
+                default -> plugin.getLogger().info(ChatColor.RED + "You provided too many arguments!");
+            }
+        }
+        return true;
     }
 }
Index: src/main/resources/plugin.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/main/resources/plugin.yml b/src/main/resources/plugin.yml
--- a/src/main/resources/plugin.yml	(revision 4c37610bb705f2517786bec0a138383cf700b42e)
+++ b/src/main/resources/plugin.yml	(revision 2e946039d35b259313f26d7e82f53fa153aec287)
@@ -15,6 +15,10 @@
       - rank
       - setr
     description: Grants a rank to players
+  setsuffix:
+    aliases:
+      - suffix
+    description: change suffix
 depend:
   - LuckPerms
 permissions:
